### ISSUE TITLE
CA2252 PreviewFeatureAnalyzer analyzer should account GenerateRequiresPreviewFeaturesAttribute property

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn-analyzers#5722]
### Analyzer

**Diagnostic ID**: [CA2252 ](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA2252 ): `Opt in to preview features before using them`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)
**Version**: [SDK 6.0.100](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug
Related to https://github.com/dotnet/runtime/discussions/61399?sort=old#discussioncomment-1624039

The MSBuild property `GenerateRequiresPreviewFeaturesAttribute` is disables adding the assemble level `[RequiresPreviewFeatures]` attribute when the project is opted in for preview feature using `<EnablePreviewFeatures>True</EnablePreviewFeatures>` property. 

[More Info](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#generaterequirespreviewfeaturesattribute)

Apparently accessing any preview features from within such a project should not cause an error

cc @terrajobst @jeffhandley 

### Steps To Reproduce

1. Create a console app targeting 6.0
2. Add following properties into the project settings
  ```xml
    <EnablePreviewFeatures>True</EnablePreviewFeatures>
    <GenerateRequiresPreviewFeaturesAttribute>False</GenerateRequiresPreviewFeaturesAttribute>
  ```
3. Use any Preview API from libraries 
4. See an error

### Expected behavior
Should not see an error for using Preview features
### Actual behavior
Seeing error for using Preview features

**Comment by terrajobst** [comment]
The intention of the design was that if `<GenerateRequiresPreviewFeaturesAttribute>` is to false, then no assembly level attribute is generated. In this case, the preview analyzer does flag every use of a preview API because the expectation is that instead of using a single assembly-level attribute, the project is opting into preview features area-by-area, by selectively applying the `[RequiresPreviewFeatures]` attribute on specific types or members.

Therefore, I'd say the current behavior is by-design.

**Comment by buyaa-n** [comment]
Thanks, apparently, I misunderstood, the `<EnablePreviewFeatures>True</EnablePreviewFeatures>` property is not only for adding assembly-level attribute
